### PR TITLE
translate strings

### PR DESCRIPTION
**Pull request by yashodha**
Overview
----------------------------------------
This adds in ts function for some strings that weren't translated.

Before
----------------------------------------
Strings not translated

After
----------------------------------------
Strings translated

### CODE REVIEW

**Comment by demeritcowboy** [comment]
It's maybe a judgement call but I'm not sure some of these SHOULD be translated. See, https://docs.civicrm.org/dev/en/latest/translation/database/, in particular:


> We try to reduce the number of strings that will be practically never seen by administrators. Very unlikely error messages should be left untranslated. [...] Administrators are very unlikely to see such strings. If they do, they will probably need the original English string in order to get support on the forums. They are also strings that are very hard to translate because of lack of context.

e.g. something like `wow, session is not an object?` seems very unlikely to ever come up.

**Comment by mlutfy** [comment]
I agree, and in general, users should not see fatal errors, we should do proper error handling when possible, i.e. if it's likely that the user will see the error.

It probably makes sense to translate these strings:

* "Please file an issue with the backtrace"
* "You do not have permission to execute this url"

And if we do translate those strings, please add a comma at the end (helps translators to know that it's not a partial sentence, and also helps for accessibility).

Unless you encountered a situation where users were seeing the other strings, I would not wrap them in ts.

**Comment by eileenmcnaughton** [comment]
@mlutfy these are all places where we should be replacing the fatal calls with throw new CRM_Core_Exception - remind me the rule as to when we do & don't translate them? Some of those look like 'should never be hit' lines & I know @colemanw is a little protective of our translators

File: CRM/Core/Invoke.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ public static function hackMenuRebuild($args) {
         return CRM_Utils_System::redirect();
       }
       else {
-        CRM_Core_Error::fatal('You do not have permission to execute this url');
+        CRM_Core_Error::fatal(ts('You do not have permission to execute this url'));
```
Comment (totten) [inline CRM/Core/Invoke.php]: If the strings are going to be translated, then I think it's good to do a little copy-edit (more consistent/correct grammar). I'll just toss in a few suggestions.

```suggestion
        CRM_Core_Error::fatal(ts('You do not have permission to execute this URL.'));
```

File: CRM/Core/BAO/CustomValueTable.php [hunk 2]
```diff
@@ -420,7 +420,7 @@ public static function &getEntityValues($entityID, $entityType = NULL, $fieldIDs
     if (!$entityID) {
       // adding this here since an empty contact id could have serious repurcussions
       // like looping forever
-      CRM_Core_Error::fatal('Please file an issue with the backtrace');
+      CRM_Core_Error::fatal(ts('Please file an issue with the backtrace'));
```
Comment (totten) [inline CRM/Core/BAO/CustomValueTable.php]:
```suggestion
      CRM_Core_Error::fatal(ts('Please file an issue with the backtrace.'));
```

File: CRM/Core/BAO/UFMatch.php [hunk 3]
```diff
@@ -62,7 +62,7 @@ public static function synchronize(&$user, $update, $uf, $ctype, $isLogin = FALS
     $userSystem = CRM_Core_Config::singleton()->userSystem;
     $session = CRM_Core_Session::singleton();
     if (!is_object($session)) {
-      CRM_Core_Error::fatal('wow, session is not an object?');
+      CRM_Core_Error::fatal(ts('wow, session is not an object?'));
```
Comment (totten) [inline CRM/Core/BAO/UFMatch.php]: ### 😜

File: CRM/Core/JobManager.php [hunk 4]
```diff
@@ -181,7 +181,7 @@ private function _getJobs() {
    */
   private function _getJob($id = NULL, $entity = NULL, $action = NULL) {
     if (is_null($id) && is_null($action)) {
-      CRM_Core_Error::fatal('You need to provide either id or name to use this method');
+      CRM_Core_Error::fatal(ts('You need to provide either id or name to use this method'));
```
Comment (totten) [inline CRM/Core/JobManager.php]:
```suggestion
      CRM_Core_Error::fatal(ts('You need to provide either an ID or name to use this method.'));
```

**Comment by eileenmcnaughton** [comment]
In terms of scope - I think it's helpful when we are removing fatals not to scope creep too far into fixing translations because I think when adding ts the sort of scope creep @totten is doing is sadly necessary

**Comment by mlutfy** [comment]
I scope-creep only the sentences that need 'ts'. Presumably this PR was motivated by a user who stumbled on some of those error messages. And if we are going to ask translators to do extra work, I think it's fair that the developers think about it a bit more ;-)

but I would only wrap in `ts` the two strings I pointed out. I'm a bit confused by some of Tim's suggestions :)

**Comment by colemanw** [comment]
I don't understand the upside of this PR. The downside is extra work for translators, and error messages that are harder to google & share in forums. What's the upside?

**Comment by totten** [comment]
I made the same presumption as @mlutfy - that the PR was motivated by a user who stumbled on some of the messages organically. (Maybe all of the messages were raised by a user... or maybe it was just 1-2 and the rest were sort of pre-emptive.)

The main issue here is the subjectivity in determining whether a message is better in its original/untranslated form or in translated form. (*Who's going to see it? Is it likely to come up? Is the content important? Will it actionable?*) I think it's a given that there's a list of messages for which any two people would evaluate the same message differently.

IMHO, the ways to engage this are either:

1. **Stick to the subjective approach.** Refine rules of engagement. If there's a bulk-update to 5+ messages, close it pro-forma; but if someone thinks it is worth the effort to switch a particular message to `ts()` form, then give them the benefit of the doubt.
2. **Remove the need for subjectivity in the code**. Treat all fatal/exception strings as translatable, but use a different notation (e.g. `ts('Bad things!', ['error' => TRUE])`) Ex: To prevent overloading the translation queue, put all those strings in a different bucket. Ex: To inform users with different needs, the error screen should show the English form AND the localized form (and maybe also a symbolic form and a hyperlink to a wiki).

**Comment by mlutfy** [comment]
@totten Option (2) sounds great, but is much more work, and so far, we do not seem to have enough real-world requirements to justify it. I would rather stick to (1) the subjective approach.

**Comment by eileenmcnaughton** [comment]
@yashodha should we close this - it feels like it got too hard. I feel like the upshot is we should leave error messages untranslated if we never expect them to be hit (but convert them to exceptions).

Where we find fatals on the form layer we should generally convert to statusBounces & those should be translated

**Comment by yashodha** [comment]
@eileenmcnaughton Agreed, I am closing this.

**Pull request closed without merging**